src/ops/project.rs

```diff
@@ -1,23 +1,82 @@
+use nom_sql::ArithmeticOperator;
+use std::fmt;
 use std::collections::HashMap;
 
 use flow::prelude::*;
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum ProjectExpressionBase {
+    Column(usize),
+    Literal(DataType),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ProjectExpression {
+    op: ArithmeticOperator,
+    left: ProjectExpressionBase,
+    right: ProjectExpressionBase,
+}
+
+impl ProjectExpression {
+    pub fn new(
+        op: ArithmeticOperator,
+        left: ProjectExpressionBase,
+        right: ProjectExpressionBase,
+    ) -> ProjectExpression {
+        ProjectExpression {
+            op: op,
+            left: left,
+            right: right,
+        }
+    }
+}
```

Comment:
Hmm, I suspect we only need these (as opposed to using `nom-sql`'s almost-identical `ArithmeticExpression`) because of the `Literal(DataType)` in the base case, which in `nom-sql` is `Scalar(Literal)`. That's a bit suboptimal -- could we use the `nom-sql` structure here since we implement `Into<DataType>` for  `nom_sql::Literal`? (I guess we'd potentially be doing the conversion every time we access the literal if we're not careful, but maybe that can be solved without duplicating the data structure.)

Reply:
On further thought, I guess another difference is that the `Column` variant of the `ProjectExpressionBase` enum holds a numeric column index, not a `Column` object. Hrm, maybe we need the duplicate structures after all.